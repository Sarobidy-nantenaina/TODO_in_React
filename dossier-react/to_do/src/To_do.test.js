import React from 'react';
import { render, unmountComponentAtNode } from 'react-dom';
import { act } from 'react-dom/test-utils';
import TodoList from './To_do';

let container;

beforeEach(() => {
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

it('adds a todo', () => {
  act(() => {
    render(<TodoList />, container);
  });
  const input = container.querySelector('input[type="text"]');
  const form = container.querySelector('form');
  act(() => {
    input.value = 'Learn Jest';
    form.dispatchEvent(new Event('submit', { bubbles: true }));
  });
  expect(container.textContent).toContain('Learn Jest');
});

it('moves a todo to done', () => {
  act(() => {
    render(<TodoList />, container);
  });
  const input = container.querySelector('input[type="text"]');
  const form = container.querySelector('form');
  act(() => {
    input.value = 'Learn Jest';
    form.dispatchEvent(new Event('submit', { bubbles: true }));
  });
  const checkbox = container.querySelector('input[type="checkbox"]');
  act(() => {
    checkbox.dispatchEvent(new Event('click', { bubbles: true }));
  });
  expect(container.textContent).toContain('Learn Jest');
});

