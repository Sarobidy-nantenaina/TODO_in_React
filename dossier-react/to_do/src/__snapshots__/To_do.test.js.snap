// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList adds todo to the list on enter key press 1`] = `ShallowWrapper {}`;

exports[`TodoList component renders TodoList component 1`] = `
<div>
  <div
    class="container"
  >
    <form
      class="form-input"
    >
      <input
        class="input-text"
        placeholder="  tap here the todo..."
        type="text"
        value=""
      />
    </form>
    <div
      class="todo-done"
    >
      <div
        class="table-todo"
      >
        <h3>
          Todo
        </h3>
        <ul />
      </div>
      <div
        class="table-done"
      >
        <h3>
          Done
        </h3>
        <ul />
      </div>
    </div>
  </div>
</div>
`;

exports[`TodoList renders correctly 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <form
      class="form-input"
    >
      <input
        class="input-text"
        placeholder="  tap here the todo..."
        type="text"
        value=""
      />
    </form>
    <div
      class="todo-done"
    >
      <div
        class="table-todo"
      >
        <h3>
          Todo
        </h3>
        <ul />
      </div>
      <div
        class="table-done"
      >
        <h3>
          Done
        </h3>
        <ul />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
